src/evaluation/move_ordering.h

```diff
@@ -60,19 +60,31 @@ class MoveOrdering {
         /* primitive insertion sort */
         for (size_t i = 1; i < moves.count(); ++i) {
             movegen::Move key = moves[i];
-            int keyScore = moveScore(board, key, ply, ttMove);
-            size_t j = i;
-            while (j > 0 && moveScore(board, moves[j - 1], ply, ttMove) < keyScore) {
-                moves[j] = moves[j - 1];
-                --j;
+
+            if (board.player == PlayerWhite) {
+                int keyScore = moveScore<PlayerWhite>(board, key, ply, ttMove);
+                size_t j = i;
+                while (j > 0 && moveScore<PlayerWhite>(board, moves[j - 1], ply, ttMove) < keyScore) {
+                    moves[j] = moves[j - 1];
+                    --j;
```

Comment:
Maybe this could be added to a lambda? Not sure if it will add runtime overhead - might need to godbolt to see if compiler can optimize it. Otherwise, fine with me :+1:

Reply:
Now that you mention it, I think the neatest solution here is another template and helper. It's also more optimised than the above lines, because the conditional for colour is entirely out of the loops over moves.